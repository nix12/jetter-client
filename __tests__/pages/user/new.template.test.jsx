import React from 'react';
import { shallow } from 'enzyme';

import Template from '../../../templates/user/new.template';

describe('<New /> template', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      onRegister: jest.fn(),
      router: {
        push: jest.fn()
      },
      error: null
    };

    wrapper = shallow(<Template {...props} />);
  });

  it('should render New component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call inputChangedHandler to update state', () => {
    const mockedEvent = { target: { value: 'abc' } };

    wrapper.instance().inputChangedHandler(mockedEvent, 'username');
    expect(wrapper.state().controls.username.value).toEqual('abc');
  });

  it('should call submitHandler to register and direct new users', () => {
    const mockEvent = { preventDefault: jest.fn() };

    return wrapper
      .instance()
      .submitHandler(mockEvent)
      .then(() => {
        expect(props.onRegister).toBeCalled();
        expect(props.router.push).toBeCalled();
      });
  });

  it('should render with no error', () => {
    expect(props.error).toEqual(null);
  });

  it('should render with error', () => {
    const mockEvent = { preventDefault: jest.fn() };

    props = {
      onRegister: jest.fn(),
      router: {
        push: jest.fn()
      },
      error: {
        username: 'error in username'
      }
    };

    wrapper = shallow(<Template {...props} />);

    return wrapper
      .instance()
      .submitHandler(mockEvent)
      .then(() => {
        expect(props.error).toEqual({
          username: 'error in username'
        });
        expect(wrapper.find('p').html()).toEqual(
          '<p>Username error in username.</p>'
        );
      });
  });
});
