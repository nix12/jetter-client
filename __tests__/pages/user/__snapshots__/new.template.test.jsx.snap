// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<New /> template should render New component 1`] = `ShallowWrapper {}`;
